src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -19,6 +19,22 @@ internal sealed class ExceptionProcessor : BaseProcessor<Activity>
     public ExceptionProcessor()
     {
 #if NET6_0_OR_GREATER || NETFRAMEWORK
+        bool getExceptionPointersSupported = true;
+
+        try
```

Comment:
Do we need this here? can we handle it here - https://github.com/open-telemetry/opentelemetry-dotnet/blob/77ef12327f720ca3defd0c9590c0197cceb5952e/src/OpenTelemetry/Trace/Builder/TracerProviderBuilderSdk.cs#L169-L176

We could catch specific exception and rethrow with custom message if that's the intention.

Reply:
@Yun-Ting If you modify this ctor to just do this...

```diff
#if NET6_0_OR_GREATER || NETFRAMEWORK
+       Marshal.GetExceptionPointers(); // Note: Attempt to access pointers to test for platform support
        this.fnGetExceptionPointers = Marshal.GetExceptionPointers;
#else
```

Then that code @vishweshbankwar shared _should_ do the rest of the work.

Reply:
Updated as suggested. Also, I removed try/catch logic for the below netstandard path for consistency (as was mentioned here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5374#discussion_r1506892321 that `Marshal.GetExceptionPointers` is internal detail that may not be helpful for the end user.